File: AoC2020/solutions/06/solution.jl
#Advent of Code 2020 day 6
input_path = (@__DIR__) * "/input"

function load(file_path=input_path)
    # Read data from file
    data = read((@__DIR__) * "/input", String)
    # Split into lines
    groups = split(strip(data), "\n\n")
    # Split groups into individuals
    return [split(group) for group in groups]
end

function task1(data)
    groups_unique = [Set(join(group)) for group in data]
    return sum(length(x) for x in groups_unique)
end

function task2(data)
    # Create a set for every member in group
    groups_members = [[Set(line) for line in group] for group in data]
    # Create intersection for every group, between all the members of the group
    intersection = [intersect(x...) for x in groups_members]
    return sum(length(x) for x in intersection)
end


function solution()
    data = load()
    println("First:  $(task1(data))")
    println("Second: $(task2(data))")
end

solution()

File: AoC2020/solutions/06/solution.py
#Advent of Code 2020 day 6
from util import *
DAY = 6
YEAR = 2020


def get_data():
    return get_input(DAY, YEAR).strip().split('\n\n')


def solution():
    data = get_data()
    groups = [set("".join(x.split())) for x in data]        # Checks for number of unique elements in a group
    members = [[set(y) for y in x.split()] for x in data]   # Check the number of unique elements per person in a group
    intersect = [j[0].intersection(*j) for j in members]    # Looks for number of unique elements in itersection of group
    first = sum(len(x) for x in groups)
    second = sum(len(x) for x in intersect)

    print(f"First:  {first}")
    print(f"Second: {second}")

solution()
